api/drivers/aws/driver.js

```diff
@@ -257,95 +257,26 @@ class AWSDriver extends Driver {
                   return reject(err);
                 }
 
-                if (image.password === null) {
-                  return this._client.ec2
-                    .waitFor('passwordDataAvailable', {
-                      InstanceId: server.id
-                    }, (err, res) => {
-
-                      if (err) {
-                        return reject(err);
-                      }
-                      return resolve({
-                        server: server,
-                        password: res.PasswordData,
-                        image: image
-                      });
-                    });
-                }
-
-                this._client.ec2
-                  .waitFor('instanceStatusOk', {
-                    InstanceIds: [server.id]
-                  }, (err) => {
-
-                    if (err) {
-                      return reject(err);
-                    }
-
-                    this._client.ec2
-                      .waitFor('systemStatusOk', {
-                        InstanceIds: [server.id]
-                      }, (err) => {
-                        if (err) {
-                          return reject(err);
-                        }
-
-                        return resolve({
-                          server: server,
-                          password: image.password,
-                          image: image
-                        });
-                      });
-                  });
-              });
-            })
-              .then((res) => {
-                return new Promise((resolve, reject) => {
-                  res.server.refresh((err, server) => {
-                    if (err) {
-                      reject(err);
-                    } else {
-                      res.server = server;
-                      resolve(res);
-                    }
-                  });
-                });
-              })
-              .then((res) => {
-                const server = res.server;
-                const password = res.password;
-                const image = res.image;
-                const ip = server.addresses.public[0] || server.addresses.private[0];
+                const ip = server.addresses.public[0];
```

Comment:
This should be `const ip = server.addresses.public[0] || server.addresses.private[0]`. Otherwise we would be loosing the benefice introduced in this PR https://github.com/Nanocloud/nanocloud/issues/184.